eventsourcing/src/main/java/org/axonframework/eventsourcing/DefaultSnapshotterSpanFactory.java

```diff
@@ -0,0 +1,141 @@
+     * The default values are:
+     * <ul>
+     *     <li>{@code separateTrace} defaults to {@code true}</li>
+     *     <li>{@code aggregateTypeInSpanName} defaults to {@code true}</li>
+     * </ul>
+     * The {@code spanFactory} is a required field and should be provided.
+     */
+    public static class Builder {
+        private SpanFactory builderSpanFactory;
+        private boolean builderSeparateTrace = false;
+        private boolean builderAggregateTypeInSpanName = true;
+
+        /**
+         * Sets the {@link SpanFactory} to use to create the spans. This is a required field.
+         *
+         * @param spanFactory The {@link SpanFactory} to use to create the spans.
+         * @return The current Builder instance, for fluent interfacing.
+         */
+        public Builder spanFactory(SpanFactory spanFactory) {
+            BuilderUtils.assertNonNull(spanFactory, "spanFactory may not be null");
+            this.builderSpanFactory = spanFactory;
+            return this;
+        }
+
+        /**
+         * Sets whether the creation of the snapshot should be represented by a separate trace. Defaults to {@code false}.
+         * <p>
+         * By default, the creation of the snapshot is part of the same trace as the scheduling of the snapshot. This
+         * happens by default in the framework as well, as snapshots are created after command invocation in the
+         * current thread. If you configure the {@link Snapshotter} to create snapshots in a separate thread, you might want to
+         * set this to {@code true}. The new root trace will be linked to the parent trace so it can be correlated.
+         * <p>
+         *
+         * @param separateTrace Whether the creation of the snapshot should be represented by a separate trace.
+         * @return The current Builder instance, for fluent interfacing.
+         */
+        public Builder separateTrace(boolean separateTrace) {
+            this.builderSeparateTrace = separateTrace;
+            return this;
+        }
+
+        /**
+         * Whether the aggregate type should be included in the span name. Defaults to {@code true}.
+         *
+         * @param aggregateTypeInSpanName Whether the aggregate type should be included in the span name.
+         * @return The current Builder instance, for fluent interfacing.
+         */
+        public Builder aggregateTypeInSpanName(boolean aggregateTypeInSpanName) {
+            this.builderAggregateTypeInSpanName = aggregateTypeInSpanName;
+            return this;
+        }
+
+        /**
+         * Initializes a {@link DefaultSnapshotterSpanFactory} as specified through this Builder.
+         *
+         * @return The {@link DefaultSnapshotterSpanFactory} as specified through this Builder.
+         */
+        public DefaultSnapshotterSpanFactory build() {
+            BuilderUtils.assertNonNull(builderSpanFactory, "spanFactory may not be null");
```

Comment:
The `validate` method is present in the other builders and invoked in the component's constructor deliberately.
This is done to ensure that _when_ somebody overrides the component, we can still protect them from missing fields in the base class.
If we move the validation to the `build()` method, users can override said method and completely disregard this protective measure.

Reply:
Alright adjusted. They can still override the validate() method though.

Reply:
Entirely true! Just given you an explanation why we went with this format that is used by *all* Axon Framework components. Wouldn't want this component to be an outlier tbh.